core/src/main/java/jenkins/install/InstallState.java

```diff
@@ -221,7 +221,7 @@ public void initializeState() {
      * {@code <installState class="jenkins.install.InstallState$CreateAdminUser" resolves-to="jenkins.install.InstallState">}
      * @see #UNUSED_INNER_CLASSES
      */
-    public Object readResolve() {
+    protected Object readResolve() {
```

Comment:
Just make it `private`. There is no legitimate use case for overriding this—we want to use this specific implementation for all subclasses.

Reply:
Wouldn't the problem be that subclasses would override it without being able to `super.readResolve()`?